Document Prometheus support (#642)
akka/akka-platform-guide: docs-source/docs/modules/deployment/pages/operator-reference.adoc

These numbers can be useful to watch the Akka cluster state during rolling updates to make sure it is in a normal state. See xref:troubleshooting.adoc[troubleshooting deployment status] for some examples.

== Reporting Metrics to Prometheus

> Exporting metrics to Prometheus requires opening a port on your Pods for Prometheus to scrape the metrics data gathered. The Akka Operator assumes you will be using https://developer.lightbend.com/docs/telemetry/current/plugins/prometheus/prometheus.html#prometheus[Lightbend Telemetry's support for Prometheus]. You have to configure the build of your application following the Lightbend Telemetry docs. You should also make sure to enable the `http-server` exporter. Then, you need to open the prometheus port:

Reviewer: > You have to configure the build of your application following the Lightbend Telemetry docs.

Are you referring to the container port? I didn't see that mentioned in the linked telemetry docs. Should we be explicit about making sure the port is open when you create the app image? Should we update the example app to enable prometheus?

Author: I see. Both my writing here and the docs on Telemetry assume a bit too much knowledge about the Prometheus architecture. I'll amend with a bit more explanation.

.kubernetes/shopping-cart-service-cr.yml
[source,yaml]
----
apiVersion: akka.lightbend.com/v1
kind: AkkaMicroservice
metadata:
  name: shopping-cart-service
spec:

>   prometheusPort: "9001"

Reviewer: Curious why this is a string. I guess because it ends up as an annotation value anyway?

Author: All `xxxPort` we have are a String which defaults to `"off"`. 😄

These numbers can be useful to watch the Akka cluster state during rolling updates to make sure it is in a normal state. See xref:troubleshooting.adoc[troubleshooting deployment status] for some examples.

== Reporting Metrics to Prometheus

> Exporting metrics to Prometheus requires opening a port on your Pods. The https://prometheus.io/docs/introduction/overview/#architecture[Prometheus Architecture documentation  {tab-icon}, window="tab"] explains all the components of Prometheus and how data is extracted from your process: for Prometheus to gather the metrics from your process you have to expose an HTTP server. The Akka Operator assumes you will be using {cinnamon-docs}/plugins/prometheus/prometheus.html#prometheus[Lightbend Telemetry's support for Prometheus]. Once you configure the build of your application following the Lightbend Telemetry docs, you should have the `http-server` exporter enabled on your configuration. That `cinnamon.prometheus.exporters += http-server` setting binds, by default, to the port `9001`. Then, you need to configure the kubernetes resource to open the prometheus port:

Reviewer 2: ```suggestion
Exporting metrics to Prometheus requires opening a port on your Pods. The https://prometheus.io/docs/introduction/overview/#architecture[Prometheus Architecture documentation  {tab-icon}, window="tab"] explains all the components of Prometheus and how data is extracted from your process: for Prometheus to gather the metrics from your process you have to expose an HTTP server. The Akka Operator assumes your application uses {cinnamon-docs}/plugins/prometheus/prometheus.html#prometheus[Lightbend Telemetry's support for Prometheus]. Once you configure the build of your application following the Lightbend Telemetry docs, you should have the `http-server` exporter enabled on your configuration. That `cinnamon.prometheus.exporters += http-server` setting binds, by default, to the port `9001`. Then, you need to configure the Kubernetes resource to open the Prometheus port:
```
